### DISCUSSION TITLE
How to have integer array as read only field from struct?

### DISCUSSION THREAD

**Discussion by tienviitta** [wjakob/nanobind#177] [Q&A] [answered]
For example: How to have read only field `crcReg` from this struct `Crc` to the binding?:
```
typedef struct Crc {
  ...
  uint32_t crcReg[4];
  ...
} Crc;
```
I have tried `.def_ro(...)` but obviously it is not working. Should I make some kind of "getter" for this and have it as `nb::ndarray<...>`?

**Comment by wjakob** [comment, marked as answer]
Just like in pybind11, there isn't great support for this. You can use `std::array` instead (with some limitations due to the type caster) or expose the underlying storage through a property involving a type like `nb::ndarray` (which can be made to return a NumPy array view).

**Reply by tienviitta** [reply]
Thank you, this is some legacy code, so hard to get rid of but I'll take that nb::ndarray approach under work.
